live/src/main/java/org/dbpedia/extraction/live/feeder/AllPagesFeeder.java

```diff
@@ -0,0 +1,99 @@
+package org.dbpedia.extraction.live.feeder;
+
+import org.dbpedia.extraction.live.core.LiveOptions;
+import org.dbpedia.extraction.live.queue.LiveQueueItem;
+import org.dbpedia.extraction.live.queue.LiveQueuePriority;
+import org.dbpedia.extraction.live.util.DateUtil;
+import org.dbpedia.extraction.live.util.ExceptionUtil;
+import org.dbpedia.extraction.util.Language;
+import org.dbpedia.extraction.util.WikiApi;
+import scala.Tuple3;
+import scala.collection.immutable.Seq;
+import scala.xml.Node;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLEncoder;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * @author Lukas Faber, Stephan Haarmann, Sebastian Serth
+ *         date 02.07.2016.
+ */
+public class AllPagesFeeder extends Feeder {
+
+    private final Language language = Language.apply(LiveOptions.options.get("language"));
+    private WikiApi api;
+    private boolean isFinished = false;
+    private String continueString = "-||";
+    private String continueTitle = "";
+    private ArrayList<Integer> allowedNamespaces = new ArrayList<>();
+    private int currentNamespace = 0;
+
+    public AllPagesFeeder(String feederName, LiveQueuePriority queuePriority, String defaultStartTime,
+                          String folderBasePath) {
+        super(feederName, queuePriority, defaultStartTime, folderBasePath);
+        try {
+            api = new WikiApi(new URL(language.apiUri()), language);
+        } catch (MalformedURLException exp) {
+            logger.error(ExceptionUtil.toString(exp), exp);
+        }
+
+        for (String namespace : LiveOptions.options.get("feeder.allpages.allowedNamespaces").split("\\s*,\\s*")) {
+            allowedNamespaces.add(Integer.parseInt(namespace));
+        }
+    }
+
+    @Override
+    protected void initFeeder() {
+
+    }
+
+    @Override
+    protected Collection<LiveQueueItem> getNextItems() {
+        ArrayList<LiveQueueItem> queue = new ArrayList<LiveQueueItem>();
+        if (!isFinished) {
+            List<Node> pageList = queryAllPagesAPI();
+            for (Node page : pageList) {
+                //queue.add(new LiveQueueItem(Long.parseLong(page.$bslash$at("pageid")), page.$bslash$at("title"), DateUtil.transformToUTC(new Date()), false, ""));
```

Comment:
Thanks. I was testing the Feeder to run on the English Wikipedia to ensure we don't have any loop (and obviously, I forgot to revert that).